src/crepe/api/Button.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <functional>
+
+#include "UiObject.h"
+
+namespace crepe {
+
+/**
+ * \class Button
+ * \brief Represents a clickable UI button, derived from the UiObject class.
+ * 
+ * This class provides functionality for a button in the UI, including toggle state,
+ * click handling, and mouse hover detection. A callback function can be provided to
+ * handle button clicks.
+ */
+class Button : public UiObject {
+public:
+	/**
+     * \brief Constructs a Button with the specified game object ID and dimensions.
+     * 
+     * \param id The unique ID of the game object associated with this button.
+     * \param width The width of the button.
+     * \param height The height of the button.
+     * \param is_toggle Optional flag to indicate if the button is a toggle button. Defaults to false.
+     * \param on_click callback function that will be invoked when the button is clicked.
+     */
+	Button(game_object_id_t id, int width, int height, std::function<void()> on_click,
+		   bool is_toggle = false);
+
+	/**
+     * \brief Indicates if the button is a toggle button (can be pressed and released).
+     * 
+     * A toggle button allows for a pressed/released state, whereas a regular button
+     * typically only has an on-click state.
+     */
+	bool is_toggle = false;
```

Comment:
Dus de Button klasse heeft zelf staat maar je geeft de user niks mee in `on_click`? Wat is het nut van `is_toggle` en waarom kan de gebruiker zelf niet een boolean bijhouden in de `on_click` handler?

Reply:
ik wilden het makkelijk maken om een toggle button te maken

Reply:
Ik zag toevallig deze comment voorbij komen en ik denk dat het fijn is als wij (de engine) al een optie voor een toggle button ondersteunen (zodat de user niet zelfstandig variabelen bij hoeft te houden). Qua mij betreft dus niet functioneel veranderen.